File: GFG/Primes sum/Primes sum.java
class Solution {
    static String isSumOfTwo(int N){
        // code here
        boolean[]primes=new boolean[N];
        Arrays.fill(primes,true);
        
        for(int i=2;i*i<=N;i++){
            if(primes[i]){
                for(int j=i*i;j<N;j+=i){
                    primes[j]=false;
                }
            }
        }
        
        for(int i=2;i<=N/2;i++){
            if(primes[i] && primes[N-i]){
                return "Yes";
            }
        }
        return "No";
    }
}

File: GFG/Primes sum/Primes sum.cpp
class Solution {
    public:
      bool checkPrime(int n)

    {

        if(n<2)

        return false;

        for(int i=2; i<=sqrt(n); i++)

        {

            if(n%i==0)

            return false;

        }

        return true;

    }
    
    string isSumOfTwo(int N){
        // code here
        
          for(int i=2; i<N; i++)

      {

          bool flag=true;

          if(!checkPrime(i))

          {

              flag=false;

          }

        if(flag==true)

        {

            if(checkPrime(N-i))

            return "Yes";

        }

      }

    return "No";
    }

};
